src/vcsparser.core/bugdatabase/BugDatabaseDllLoader.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+
+namespace vcsparser.core.bugdatabase
+{
+    public interface IBugDatabaseDllLoader
+    {
+        IBugDatabaseProvider Load(string path, IEnumerable<string> args, IWebRequest webRequest);
+    }
+
+    public class BugDatabaseDllLoader : IBugDatabaseDllLoader
+    {
+        private readonly ILogger logger;
+        private readonly IBugDatabaseFactory bugDatabaseFactory;
+
+        public BugDatabaseDllLoader(ILogger logger, IBugDatabaseFactory bugDatabaseFactory)
+        {
+            this.logger = logger;
+            this.bugDatabaseFactory = bugDatabaseFactory;
+        }
+
+        public IBugDatabaseProvider Load(string path, IEnumerable<string> args, IWebRequest webRequest)
+        {
+            _Assembly dll;

```

Comment:
Why instead of returning null and treating the error up the chain you don't just throw an exception?
You could use a single try/catch that adds some extra information in the exception like "Error trying to load dll blah with args blah " + Message.

That would save you a lot of conditionals. Also as a rule of thumb if the method is > 20 lines of code, extracting in smaller methods is a good thing to do.